repo-agent/pkg/agentoutput/watcher.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package agentoutput
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	outputFile = "/workspaces/agent-output.txt"
+)
+
+// Run starts the sidecar watcher loop.
+// componentName is used for logging (e.g., "issue", "review").
+// gvr is the GroupVersionResource to update.
+func Run(componentName string, gvr schema.GroupVersionResource) {
```

Comment:
Using `os.Exit(1)` in a library function is generally discouraged. It forces the entire application to terminate. This function should return an error instead, allowing the caller to decide how to handle the failure.